app/src/main/java/io/apicurio/registry/storage/impl/sql/AbstractSqlRegistryStorage.java

```diff
@@ -1248,8 +1252,14 @@ public ArtifactSearchResultsDto searchArtifacts(Set<SearchFilter> filters, Order
                 binder.bind(countQuery, idx);
                 idx++;
             }
-            artifactsQuery.bind(idx++, limit);
-            artifactsQuery.bind(idx++, offset);
+            // TODO find a better way to swap arguments
+            if ("sqlserver".equals(sqlStatements.dbType())) {
+                artifactsQuery.bind(idx++, offset);
+                artifactsQuery.bind(idx++, limit);
+            } else {
+                artifactsQuery.bind(idx++, limit);
+                artifactsQuery.bind(idx++, offset);
+            }
```

Comment:
Agree with your TODO.  This is fine for now.  The entire search strategy needs to be evaluated.  It's a bit fragile already.